Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -35,6 +35,9 @@ extension Compiler.ByteCodeGen {
       builder.buildUnresolvedReference(id: id)
 
     case let .changeMatchingOptions(optionSequence):
+      if !builder.hasReceivedInstructions {
+        builder.initialOptions.apply(optionSequence.ast)
+      }
```

Comment:
I think it's a much cleaner model if we get the initial set by checking the tree for global options. That is, whatever creates `ByteCodeGen` struct sets them.

Reply:
That is, resolve the TODO in https://github.com/apple/swift-experimental-string-processing/blob/4f8f67a653d363d56fab477a6205481919cd6529/Sources/_StringProcessing/Compiler.swift#L29

Reply:
We discussed — going to merge this as is and look into computing this directly from `DSLTree` in the future.